courses/intro-to-spark/book/02_Flow_Analysis.rst

```diff
@@ -701,16 +718,17 @@ found.
 
        type Search_Result (Found : Boolean := False) is record
           case Found is
-          when True =>
-             Content : Integer;
-          when False => null;
+             when True =>
+                Content : Integer;
+             when False => null;
           end case;
        end record;
 
        procedure Search_Array
          (A      :     Array_Of_Positives;
           E      :     Positive;
-          Result : out Search_Result);
+          Result : out Search_Result)
+       with Pre => not Result'Constrained;
```

Comment:
I might have missed it, but I don't remember seeing `'Constrained` being mentioned or explained before this instance. If it's the first instance in the SPARK course, a short description would be nice.

Reply:
You're right. That's a technical precondition, and I was not too sure this deserved to be explained here, but I found a way to explain it without going into too much details.